pkg/chunked/internal/compression.go

```diff
@@ -167,13 +170,41 @@ func WriteZstdChunkedManifest(dest io.Writer, outMetadata map[string]string, off
 		return err
 	}
 
+	var compressedTarSplitBuffer bytes.Buffer
+	compressTarSplitWriter, err := ZstdWriterWithLevel(&compressedTarSplitBuffer, level)
+	if err != nil {
+		return err
+	}
+	if _, err := compressTarSplitWriter.Write(tarSplitData); err != nil {
+		compressTarSplitWriter.Close()
+		return err
+	}
+	if err := compressTarSplitWriter.Close(); err != nil {
+		return err
+	}
+	compressedTarSplit := compressedTarSplitBuffer.Bytes()
+
+	tarSplitDigester := digest.Canonical.Digester()
+	tarSplitChecksum := tarSplitDigester.Hash()
+	if _, err := tarSplitChecksum.Write(compressedTarSplit); err != nil {
+		return err
+	}
+	outMetadata[TarSplitChecksumKey] = tarSplitDigester.Digest().String()
+	tarSplitOffset := manifestOffset + uint64(len(compressedManifest)) + 8
```

Comment:
Would it be easy enough to get rid of the `+8` magic?

The caller of this function already has an `ioutils.WriteCounter`; it can pass that into this function, and AFAICS we could get rid of the `offset` variable entirely. Then, just call `appendZstdSkippableFrame`, and read the current offset to put into the annotation.